src/rrdp-iostat/rrdp_iostat.ml

```diff
@@ -202,21 +202,56 @@ let previous_map : (int * (int * (string * string))) list ref = ref []
 let phypath_to_sr_vdi : ((string, (string * string)) Hashtbl.t) = Hashtbl.create 20
 
 let refresh_phypath_to_sr_vdi () =
+  let iter_sr_dirs path f =
```

Comment:
Below the code is used more in a fold-like fashion. I think this should either implement a fold, or simply return the list of directories to consider. If the list of entries is long, returning a sequence would be an option.